extension/splitter/splitter.go

```
@@ -0,0 +1,55 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package splitter
+
+import (
+	"net"
+
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+
+	"github.com/ava-labs/hypersdk/extension/grpcindexer"
+
+	pb "github.com/ava-labs/hypersdk/proto/pb/externalsubscriber"
+)
+
+type Splitter struct {
+	gRPCServer *grpc.Server
+	listener   net.Listener
+	tcpPort    string
+	log        logging.Logger
+}
+
+func NewSplitter(e *grpcindexer.ExternalSubscriberServer, log logging.Logger, tcpPort string) *Splitter {
+	lis, err := net.Listen("tcp", tcpPort) // #nosec G102
+	if err != nil {
+		log.Fatal("Could not listen to TCP port", zap.Any("Address", tcpPort))
+	}
```

Can we propagate the error to the caller instead of using `log.Fatal` here?